libc/stdio_scanf.c

```diff
@@ -0,0 +1,4371 @@
+{
+	fclose(filep);
+}
+
+
+TEST(stdio_scanf_cspn, c)
+{
+	const char *format = "%corem-ips%cm-dolo%c";
+	char buff[BUFF_LEN_STR] = "Lorem-ipsum-dolor";
+	char c1, c2, c3;
+
+	fprintf(filep, "%s", buff);
+	rewind(filep);
+
+	TEST_ASSERT_EQUAL_INT(3, test_vfscanfWrapper(filep, format, &c1, &c2, &c3));
+	TEST_ASSERT_EQUAL_CHAR('L', c1);
+	TEST_ASSERT_EQUAL_CHAR('u', c2);
+	TEST_ASSERT_EQUAL_CHAR('r', c3);
+
+	rewind(filep);
+
+	TEST_ASSERT_EQUAL_INT(3, fscanf(filep, format, &c1, &c2, &c3));
+	TEST_ASSERT_EQUAL_CHAR('L', c1);
+	TEST_ASSERT_EQUAL_CHAR('u', c2);
+	TEST_ASSERT_EQUAL_CHAR('r', c3);
+
+	TEST_ASSERT_EQUAL_INT(3, test_vsscanfWrapper(buff, format, &c1, &c2, &c3));
+	TEST_ASSERT_EQUAL_CHAR('L', c1);
+	TEST_ASSERT_EQUAL_CHAR('u', c2);
+	TEST_ASSERT_EQUAL_CHAR('r', c3);
+
+	TEST_ASSERT_EQUAL_INT(3, sscanf(buff, format, &c1, &c2, &c3));
+	TEST_ASSERT_EQUAL_CHAR('L', c1);
+	TEST_ASSERT_EQUAL_CHAR('u', c2);
+	TEST_ASSERT_EQUAL_CHAR('r', c3);
+}
+
+
+TEST(stdio_scanf_cspn, c_ascii)
+{
+	char buff[BUFF_LEN_STR] = { 0 };
+	char c;
+	int i;
+
+	for (i = 1; i < 128; i++) {
+		buff[i - 1] = i;
+	}
+
+	fprintf(filep, "%s", buff);
+	rewind(filep);
+
+	for (i = 1; i < 128; i++) {
+		TEST_ASSERT_EQUAL_INT(1, test_vfscanfWrapper(filep, "%c", &c));
+		TEST_ASSERT_EQUAL_CHAR(i, c);
+
+		fseek(filep, i - 1, SEEK_SET);
+
+		TEST_ASSERT_EQUAL_INT(1, fscanf(filep, "%c", &c));
+		TEST_ASSERT_EQUAL_CHAR(i, c);
+
+		/*
+		 * This fseek is used because of issue #639
+		 * https://github.com/phoenix-rtos/phoenix-rtos-project/issues/639
+		 */
+
+#ifdef __phoenix__
+		fseek(filep, i, SEEK_SET);
+#endif
+
+		TEST_ASSERT_EQUAL_INT(1, test_vsscanfWrapper(&buff[i - 1], "%c", &c));
+		TEST_ASSERT_EQUAL_CHAR(i, c);
+
+		TEST_ASSERT_EQUAL_INT(1, sscanf(&buff[i - 1], "%c", &c));
+		TEST_ASSERT_EQUAL_CHAR(i, c);
+	}
+}
+
+
+TEST(stdio_scanf_cspn, s_path)
+{
+	char buff[BUFF_LEN_STR] = TESTFILE_PATH;
+	char res[BUFF_LEN_STR];
+
+	fprintf(filep, "%s", buff);
+	rewind(filep);
+
+	TEST_ASSERT_EQUAL_INT(1, test_vfscanfWrapper(filep, "%s", res));
+	TEST_ASSERT_EQUAL_STRING(TESTFILE_PATH, res);
+
+	rewind(filep);
+
+	TEST_ASSERT_EQUAL_INT(1, fscanf(filep, "%s", res));
+	TEST_ASSERT_EQUAL_STRING(TESTFILE_PATH, res);
+
+	TEST_ASSERT_EQUAL_INT(1, test_vsscanfWrapper(buff, "%s", res));
+	TEST_ASSERT_EQUAL_STRING(TESTFILE_PATH, res);
+
+	TEST_ASSERT_EQUAL_INT(1, sscanf(buff, "%s", res));
+	TEST_ASSERT_EQUAL_STRING(TESTFILE_PATH, res);
+}
+
+
+TEST(stdio_scanf_cspn, s_pick)
+{
+	char buff[BUFF_LEN_STR] = TEST_STR;
+	char firstWord[MAX_TESTSTR_WORDLEN];
+	char secondWord[MAX_TESTSTR_WORDLEN];
+	char thirdWord[MAX_TESTSTR_WORDLEN];
+	char fourthWord[MAX_TESTSTR_WORDLEN];
+	char decimalWord[MAX_TESTSTR_WORDLEN];
+	char hexWord[MAX_TESTSTR_WORDLEN];
+	const char *format = "%s %s %s %s amet,Vestibulum ante ipsum primis in faucibus orci luctus %s et ultrices posuere cubilia "
+						 "curae %s";
+	fprintf(filep, "%s", buff);
+	rewind(filep);
+
+	/* clang-format off */
```

Comment:
why `clang-format off`?

if needed, please at least fix style by hand to be as compliant as possible

Reply:
Bit of leftovers after the cleanup situation with formats (Used because of too long format specifiers).